internal/certgen/certgen.go

```diff
@@ -60,11 +69,15 @@ type Config struct {
 	OutputPEM bool
 }
 
-// ContourCerts holds all three keypairs required, the CA, the Contour, and Envoy keypairs.
-type ContourCerts struct {
-	CA      *keyPair
-	Contour *keyPair
-	Envoy   *keyPair
+type certData struct {
+	Filename string
+	Data     []byte
+}
+
+// writePEM writes a certificate out to its filename in outputDir.
+func (cd *certData) writePEM(outputDir string) error {
+	return writePEMFile(outputDir+"/"+cd.Filename, cd.Data)
```

Comment:
nit: although we don't support and proably never will support building for windows, its probably not a bad idea to future proof this with path/filepath.join(outputDir, cd.Filename). If for no other reason than someone will probably add this later and why have to review two PRs, eh?